src/edu/stanford/rsl/tutorial/weightedtv/Perform2DwTV.java

```diff
@@ -0,0 +1,309 @@
+package edu.stanford.rsl.tutorial.weightedtv;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import ij.IJ;
+import ij.ImageJ;
+import ij.ImagePlus;
+import edu.stanford.rsl.tutorial.filters.RamLakKernel;
+import edu.stanford.rsl.conrad.data.numeric.Grid1D;
+import edu.stanford.rsl.conrad.data.numeric.Grid2D;
+import edu.stanford.rsl.conrad.data.numeric.NumericPointwiseOperators;
+import edu.stanford.rsl.conrad.geometry.Rotations;
+import edu.stanford.rsl.conrad.geometry.transforms.ScaleRotate;
+import edu.stanford.rsl.conrad.numerics.SimpleMatrix;
+import edu.stanford.rsl.conrad.utils.ImageUtil;
+import edu.stanford.rsl.tutorial.phantoms.Phantom;
+import edu.stanford.rsl.tutorial.phantoms.SheppLogan;
+import edu.stanford.rsl.tutorial.fan.FanBeamProjector2D;
+import edu.stanford.rsl.tutorial.fan.FanBeamBackprojector2D;
+
+/**
+ *This is an example to apply weighted total variation (wTV) on 2D fan-beam limited angle tomography.
+ *wTV is according to [1].
+ *[1]Candes. Enhancing Sparsity by Reweighted l1 Minimization
+ * @author Yixing Huang
+ *
+ */
+public class Perform2DWeightedTV {
+	static boolean isOpencl=true;
+
+	 static int maxIter = 500;
+	 static int maxTVIter = 10;
+	 static double epsilon = 1.0e-10;
+	 static double error = 1.0;
+	 static double error2;
+	 static int iter = 0;
+	 static double step=0;
+	
+	//SART
+	static Grid2D phan;
+	static Grid2D sinogram;
+	static Grid1D sinoDiff1D;
+	static Grid2D recon;
+	static Grid2D reconPre;
+	static Grid2D normSinogram;
+	static Grid2D normGrids;
+	static Grid2D localImageUpdate;
+	//weighted TV
+	static TVGradient TVGrad;
+	static Grid2D TVGradient;
+	//Geometry
+	static int imgSizeX;
+	static int imgSizeY;
+	static double gammaM;
+	static double maxT;
+	static double deltaT;
+	static double focalLength;
+	static double maxBeta;
+	static double deltaBeta;
+	static int spacingX=1,spacingY=1;
+	static FanBeamProjector2D projector;
+	static FanBeamBackprojector2D backProj;
+	
+	public static void main(String[] args) throws IOException {
+		new ImageJ();
+		
+		imgSizeX = 256;
+		imgSizeY=imgSizeX;
+		phan = new SheppLogan(imgSizeX, false);
+		phan.setSpacing(spacingX,spacingY);
+		phan.show("The Phantom");
+	   	
+	   
+	   initialGeometry();
+	   initialSART();
+	   createNormProj();
+       createNormGrids();	
+       
+	
+       //initial wTV
+		TVGradient= new Grid2D(imgSizeX, imgSizeY);
+		TVGradient.setSpacing(1, 1);
+		TVGrad = new TVGradient(recon);
+		TVGrad.initialWeightMatrix();
+   
+
+		
+		while (error > epsilon && iter <=maxIter) {
+		 sartIterate();		 
+		 weightedTVIterate();
+
+		 TVGrad.weightMatrixUpdate(recon);
+		 outPutError();
+		 iter++;
+		}
+
+		recon.show("Reconstructed Image");
+	}
+	
+	
+	
+/**
+ * SART iteration part
+ */
+private static void sartIterate(){
```

Comment:
WHy is this method static? Shouldn't it be a non static method? This also holds for all the other methods...
I think your methods should be part of the object..,